apps/examples/testcase/ta_tc/media/utc/utc_media_bufferoutputdatasource.cpp

```diff
@@ -0,0 +1,185 @@
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_media_BufferOutputDataSource_setSampleRate_p(void)
+{
+	BufferOutputDataSource dataSource((OnBufferDataReached)callback);
+	unsigned int compare_sampleRate = 32000;
+	dataSource.setSampleRate(compare_sampleRate);
+	TC_ASSERT_EQ("utc_media_BufferOutputDataSource_setSampleRate", dataSource.getSampleRate(), compare_sampleRate);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_media_BufferOutputDataSource_setPcmFormat_p(void)
+{
+	BufferOutputDataSource dataSource((OnBufferDataReached)callback);
+	media::audio_format_type_t compare_pcmFormat = media::AUDIO_FORMAT_TYPE_S8;
+	dataSource.setPcmFormat(media::AUDIO_FORMAT_TYPE_S8);
+	TC_ASSERT_EQ("utc_media_BufferOutputDataSource_setPcmFormat", dataSource.getPcmFormat(), compare_pcmFormat);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_media_BufferOutputDataSource_CopyConstructor_p(void)
+{
+	BufferOutputDataSource dummy((OnBufferDataReached)callback);
+	BufferOutputDataSource dataSource(dummy);
+
+	TC_ASSERT_EQ("utc_media_BufferOutputDataSource_CopyConstructor", dataSource.getChannels(), dummy.getChannels());
+	TC_ASSERT_EQ("utc_media_BufferOutputDataSource_CopyConstructor", dataSource.getSampleRate(), dummy.getSampleRate());
+	TC_ASSERT_EQ("utc_media_BufferOutputDataSource_CopyConstructor", dataSource.getPcmFormat(), dummy.getPcmFormat());
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_media_BufferOutputDataSource_EqualOperator_p(void)
+{
+	BufferOutputDataSource dummy((OnBufferDataReached)callback);
+	BufferOutputDataSource dataSource = dummy;
+
+	TC_ASSERT_EQ("utc_media_BufferOutputDataSource_EqualOperator", dataSource.getChannels(), dummy.getChannels());
+	TC_ASSERT_EQ("utc_media_BufferOutputDataSource_EqualOperator", dataSource.getSampleRate(), dummy.getSampleRate());
+	TC_ASSERT_EQ("utc_media_BufferOutputDataSource_EqualOperator", dataSource.getPcmFormat(), dummy.getPcmFormat());
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_media_BufferOutputDataSource_open_p(void)
+{
+	BufferOutputDataSource dataSource((OnBufferDataReached)callback);
+
+	TC_ASSERT_EQ("utc_media_BufferOutputDataSource_open", dataSource.open(), true);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_media_BufferOutputDataSource_close_p(void)
+{
+	BufferOutputDataSource dataSource((OnBufferDataReached)callback);
+
+	TC_ASSERT_EQ("utc_media_BufferOutputDataSource_close", dataSource.close(), true);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_media_BufferOutputDataSource_isPrepare_p(void)
+{
+	BufferOutputDataSource dataSource((OnBufferDataReached)callback);
+
+	TC_ASSERT_EQ("utc_media_BufferOutputDataSource_isPrepare", dataSource.isPrepare(), true);
+	TC_SUCCESS_RESULT();
+}
+
+
+static void utc_media_BufferOutputDataSource_write_p(void)
+{
+	BufferOutputDataSource dataSource((OnBufferDataReached)callback);
+	unsigned char dummy[] = "dummy";
+	ssize_t dummySize = 6;
+	int count = 5;
+
+	for (int i = 0; i < count; i++) {
```

Comment:
int i;
for (i=0;xxxxxxx)
Separate is recommended because some compilers not support

Reply:
FYI) `for (int i = 0;` is C99 standard. If we enable C99 as default, this code is OK.